src/Units/UnitState.java

```diff
@@ -22,10 +22,11 @@ public abstract class UnitState implements Serializable
   public Commander CO;
 
   /**
-   * HP determines the current actual durability of a unit.
-   * It's typically in range [1-10]. A unit at 0 HP is dead.
-   * Health is HP value as a percentage, thus ~10x the HP value.
-   * When determining HP, health must always be rounded up.
+   * HP/health determines the current durability of a unit.<p>
+   * It's typically in range [1,100]. A unit at 0 HP is dead.<p>
+   * HP is used to scale unit effectiveness, and normally rounds up to the next 10 for this purpose.<p>
+   * HP is also shown only in this [1,10] scale, so the last digit of the health number is considered hidden information.<p>
+   * As such, direct access of this value should only be used when you have a specific reason to choose it over getHP()<p>
```

Comment:
What is "It" that is typically in range [1,100]? `HP/health', or one or or the other? Pronouns are the worst.
You are using two apparent synonyms in an equation without defining either one (or rather, defining one of them and not specifying which).

You can't say "in this [1,10] scale" when you haven't yet introduced the scale. You do mention rounding in the previous sentence, but that is in the context of the [1,100] range.

Make sure I have this straight:
`health` is just public with no accessor, and is typically in [1,100], and represents the true/precise value.
`getHP()` returns a number in [1,100], but ceilinged to the nearest 10.
`getHPFactor()` returns a number in [1,10]? If so that's just named wrong (unless health goes to 1000%)
To get the health value displayed in the UI, we have to `getHP()/10` (or I guess `getHPFactor()`, but I refuse)

I don't like having a `getFactor()` that doesn't return a factor, or a `getHP()` that returns something different than what is called HP in the UI. I do like keeping health in the range [1,100] until it is drawn, but names matter.

Reply:
HP/health. Both. I think(?) I've settled on "GUI HP" for what we see in the GUI. Fair criticism.
The intent was to show that they are indeed synonyms. Whoops.

A number in [0,100] that doesn't have a ones digit can be encoded in [0,10]. But fair enough.

Yes.
Yes.
Yes. I would rather delete it entirely than bring floating point back.
Yes, and fair enough.

Fair enough.
I was under the impression that HP and health being different things was a source of confusion.

So... do I just rename all the functions to talk about "health" and then we bring back the health/HP divide?
And my new `getHPFactor()` can become just `getHP()`?

Reply:
We could also bring back `getPreciseHealth()`, I guess, though I kind of hate having nothing-getters.
~~clearly, it should take a string argument that justifies why you're calling it~~

Reply:
I think it's good to standardize on [0, 100] for calculations to avoid floating point nonsense. I'd just call it `health` though all around so we don't create ambiguity with `HP`, which is overloaded for the GUI context (and it feels silly to have `HP` and `GUI HP` be different things).

I'm fine with removing  `getHPFactor`; the name is ambiguous at best. We can keep `getHP` if convenient I guess? Or banish that too, update calcs that currently use it to work in terms of health instead, and just divide by 10 before drawing.